js/shopping-list.js

```diff
@@ -0,0 +1,50 @@
+var ulElement = document.querySelector('ul');
+
+function addListItems(event){
+    var form = event.target;
+    var items = form.querySelector('#items').value.split(',');
+
+    if(items[0].length > 0){
+        document.querySelector('#user-warning').textContent = "";
+        for(var i = 0; (i < items.length) && (items[i].length > 0) && (items[i] !== ' '); i++){
```

Comment:
We could use another function similar to `.map` called [Array.prototype.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) that filters your Array for any items for which your provided function does not return true or a truthy value. So, if we were to add to the other code I suggested:

![Capture](https://user-images.githubusercontent.com/20998142/84096101-db33c800-a9b5-11ea-80b0-c308f4d079be.PNG)

So we'll remove all the excess whitespace, then additionally remove any elements that, with whitespace removed, have a length of 0. ~~Then we can remove the `if` statement altogether because if `nonEmptyItems.length` is 0, the `for` loop won't execute~~The conditional for the `for` loop is also much simpler to read and understand.

EDIT: Nevermind, you'll want to leave the `if` statement in - I didn't see the `else` that sets a user error message if they didn't enter any items. However, you can change it to `if (nonEmptyItems.length > 0)`.